ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Address.java

```diff
@@ -20,14 +20,14 @@
 import org.hyperledger.besu.ethereum.rlp.RLP;
 import org.hyperledger.besu.ethereum.rlp.RLPException;
 import org.hyperledger.besu.ethereum.rlp.RLPInput;
-import org.hyperledger.besu.util.bytes.BytesValue;
-import org.hyperledger.besu.util.bytes.DelegatingBytesValue;
+
+import java.util.Objects;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.tuweni.bytes.Bytes;
 
 /** A 160-bits account address. */
-public class Address extends DelegatingBytesValue
-    implements org.hyperledger.besu.plugin.data.Address {
+public class Address implements org.hyperledger.besu.plugin.data.Address, Comparable<Address> {
```

Comment:
Is there some Tuweni equivalant of the DelegatingBytesValue?  I have mixed feelings about adding toBytes everywhere.  We had baked into our API assumptions that Addresses were just another kind of BytesValue.

Reply:
yes, that turned out to be 80% of the work. `DelegatingBytes32` exists, but we made it package visible only. It is only available via a static factory method, and not for subtyping.

I see two choices: we can go back to using Hash as a delegate bytes class, or we can rip out the Hash class. Which one do you prefer?

Reply:
Yes, same with Hash.  I had not gone that far down the java files to see that Hash also was a delegated impl.  

I would prefer the type extension.  You will want to make sure only the minimum required APIs have protected or higher access, and possibly mark some methods final.

Reply:
Brought in type extension. Let's see how it plays out.